### DISCUSSION TITLE
Change state in widget controller from repository

### DISCUSSION THREAD

**Discussion by mk0205k** [rrousselGit/riverpod#2560] [Q&A]
Hi, 
is it possible chage state in widget controller (Notifier) from repository (FutureProvider)? I have FutureProvider with webViewController with JavaScriptChannel and I want change button state when onMessageReceived.

**Comment by mafreud** [comment]
Do you want to manipulate Notifier's state from other provider(in this case, FutureProvider)?

**Reply by mk0205k** [reply]
Yes, is it possible?

**Comment by rrousselGit** [comment]
That's anti-pattern. A provider should not modify other providers during its initialization. As such as FutureProvider has no mean of updating a Notifier.

**Comment by mk0205k** [comment]
Thank you for your answer. I thought so, but wanted to be sure.
